pkg/kubelet/status/status_manager.go

```diff
@@ -352,12 +352,22 @@ func (m *manager) SetPodStatus(pod *v1.Pod, status v1.PodStatus) {
 	// Make sure we're caching a deep copy.
 	status = *status.DeepCopy()
 
+	// Set the observedGeneration for this pod status. Static pods are excluded.
+	if podObservedGenerationTrackingEnabled(pod) && !metav1.HasAnnotation(pod.ObjectMeta, v1.MirrorPodAnnotationKey) {
+		status.ObservedGeneration = pod.GetGeneration()
```

Comment:
same comment about testing what happens if the kubelet tries to set the field and the API server clears it, to avoid a hotloop / infinite repatch retry

Reply:
https://github.com/kubernetes/kubernetes/pull/130352#discussion_r1977730000